### PR TITLE
Create Votes, Tally Votes on Frontend and some bugfixes

### LINKED ISSUE #11: voting react page

### PR DESCRIPTION
**Pull request by a1719756**
Still has no authorisation because I don't really know how to do it. Also no idea how to do the role checking on navbar (I tried). Fixes some bugs with deleting and voting

Closes #11.

### CODE REVIEW

File: frontend/src/login/login.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ class loginPage extends Component {
                 (r) => {
                     if(r && r.status && r.status === 200){
                         this.props.login();
-                        this.setState({isPageLoading: false})
+                         this.setState({isPageLoading: false})
```
Comment (benedict-soh) [inline frontend/src/login/login.js:68]: Should this end in a `;`?

Comment (a1719756) [reply frontend/src/login/login.js:68]: It can but doesn't have to cos javascript I guess

**Comment by a1719756** [comment]
I added authorisation to all the pages that need it. Navbar also has different link depending on user role.

File: frontend/src/components/AuthWrapper.js [hunk 2]
```diff
@@ -72,12 +72,13 @@ export function withAuthorisation(WrappedComponent, role) {
     const mapDispatchToProps = (dispatch) => ({
         logout: () => dispatch(authActions.logout()),
         login: () => dispatch(authActions.login()),
+        setAuth: (auth) => dispatch(authActions.setAuth(auth)),
     });
```
Comment (LittenJV) [inline frontend/src/components/AuthWrapper.js:77]: just need to use this function after line 51 to set the authorisation

**Review by LittenJV — DISMISSED** [review]
nice, just set auth in authwrapper and you're ready to merge!

**Review by LittenJV — APPROVED** [review]

**Pull request merged**